### PR TITLE
libgpg-error: update configure/libtool.m4 to support macOS >= 11.x

### PR DESCRIPTION
**Pull request by aconchillo**
libgpg-error's libtool.m4 doesn't properly support macOS >= 11.x (see libtool.rb formula). This causes the library to be linked with a flat namespace which might cause issues when dynamically loading the library with dlopen under some modes, see:

  https://lists.gnupg.org/pipermail/gcrypt-devel/2021-September/005176.html

-----

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

### CODE REVIEW

File: Formula/libgpg-error.rb [hunk 1]
```diff
@@ -18,7 +19,24 @@ class LibgpgError < Formula
     sha256 x86_64_linux:  "b862b10316313dabc9ddc7774e37c41c0700b886d361b86a1485052287b73fd0"
   end
 
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "gettext" => :build
```
Comment (carlocab) [inline Formula/libgpg-error.rb:24]: Can we try building without this? In my experience this isn't actually needed except when building locally on systems that happen to have it installed.

Comment (aconchillo) [reply Formula/libgpg-error.rb:24]: I have made the patch directly to configure instead. This way we don't need to add any additional build dependencies.

File: Formula/libgpg-error.rb [hunk 2]
```diff
@@ -18,6 +19,22 @@ class LibgpgError < Formula
...
+  depends_on "libtool" => :build
+
```
Comment (carlocab) [inline Formula/libgpg-error.rb:26]:
```suggestion
```

Comment (aconchillo) [reply Formula/libgpg-error.rb]: ooops, removed.

**Comment by aconchillo** [comment]
Good news. These libtool.m4 fixes have been applied upstream:

https://git.gnupg.org/cgi-bin/gitweb.cgi?p=libgpg-error.git;a=commit;h=a3987e44970505a5540f9702c1e41292c22b69cf

So we can remove them from the formula when there's a new release.

**Comment by aconchillo** [comment]
Hi! Anything else I can do to help and get this merged? Happy to help. 😄

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**